src/pretty.rs

```diff
@@ -60,6 +60,17 @@ where
         self.text(s)
     }
 
+    fn multiline_string(&'a self, s: &str) -> DocBuilder<'a, Self, A> {
+        let percent_count = s.split(|c| c != '%').map(|v| v.len()).max().unwrap();
+        let delimiter = "%".repeat(percent_count + 1);
+        self.hardline()
+            .append(self.intersperse(
+                s.lines().map(|d| self.text(d.to_owned())),
+                self.hardline().clone(),
+            ))
+            .enclose(format!("m{delimiter}\""), format!("\"{delimiter}"))
```

Comment:
I didn't even know you could do that in Rust. Has this landed recently?

Reply:
I think it's https://github.com/rust-lang/rust/pull/90473. I guess, it was stabilized in 1.58, can't really tell easily.